docstring_parser/parser/numpy.py

```diff
@@ -0,0 +1,67 @@
+"""NumPy-style docstring parsing."""
+
+from numpydoc import docscrape
+
+from .common import ParseError, Docstring, DocstringMeta
+
+
+def parse(text: str) -> Docstring:
+    """
+    Parse the NumPy-style docstring into its components.
+
+    :returns: parsed docstring
+    """
+    ret = Docstring()
+    if not text:
+        return ret
+
+    try:
+        doc = docscrape.NumpyDocString(text)
```

Comment:
Can this be done without `numpydoc` and its heavy dependencies?

Reply:
Three options:

1. `docstring_parser` depends on `numpydoc` but not its dependencies. I'm sure this is possible with research, and is the ideal scenario
2. `docstring_parser` vendors its own `numpydoc`. This has the benefit that we can take only what we need from `numpydoc` (specifically `numpydoc.docparse`), but of course it has the usual downsides of vendoring
3. We implement our own NumPy-style docstring parsing with RegEx, like the current Google-style and Sphinx-style parsing. NumPy-style is [more complex](https://numpydoc.readthedocs.io/en/latest/format.html#sections) however, so this will take some time

Reply:
I'd go with 3

Reply:
Option 1 can be accommodated with `numpydoc` being an optional dependency (ie in the `extras_require` of the package configuration), so users can install all the dependencies if they specifically want NumPy-style docstring parsing without affecting users not needing it